src/main.cpp

```diff
@@ -115,19 +143,18 @@ void setup() {
 }
 
 void loop() {
-  static long lastFlush = 0;
-  static boolean delayedAppInit = true;
-
-  // check possible interaction with ULP program
-  loop_ulp();
 
+  mainAppSwitcher->loop(hal);
   hal->handleWakeupFromLightSleep();
   hal->checkButtons();
+  hal->fetchRtcTime();         //TODO - doesn't need every cycle update. 
```

Comment:
Is there any chance to finalize this before accepting this PR?

Reply:
Yeah that's easy but it's a trade-off we can just place it in the 30fps limiter of the screenflush. I wasn't sure which trade-off i wanted to make here. What would u suggest ?

Reply:
> This is btw the reason also why im have a PR to the develop branch :D

In that case, you could mark your PR as a draft. :grin: 

>  I wasn't sure which trade-off i wanted to make here.

Sry, but what are the tradeoffs? Fetch every cycle or... Every frame? The latter sounds better.

Reply:
if we include it in the display refresh things could break we certain apps don't update the screenbuffer as much. Simply because we fill the screen buffer first than than cal the display refresh function. We would be always a frame to late. and in the case of only 1 screenrefresh every 3 seconds for example this would break stuf.

Reply:
Ahh, got it. Okay - then it is fine there. Additionally the system should have always the newest time cached anyways.